sormas-backend/src/main/java/de/symeda/sormas/backend/caze/surveillancereport/SurveillanceReportService.java

```diff
@@ -50,24 +53,25 @@ public Predicate buildCriteriaFilter(SurveillanceReportCriteria criteria, Criter
 		return filter;
 	}
 
-	public List<SurveillanceReport> getByCaseUuid(String caseUuid) {
-		CriteriaBuilder cb = em.getCriteriaBuilder();
-		CriteriaQuery<SurveillanceReport> cq = cb.createQuery(SurveillanceReport.class);
-		Root<SurveillanceReport> sampleRoot = cq.from(SurveillanceReport.class);
-		Join<SurveillanceReport, Case> caseJoin = sampleRoot.join(SurveillanceReport.CAZE, JoinType.LEFT);
+	public List<SurveillanceReport> getByCaseUuids(List<String> caseUuids) {
 
-		cq.where(caseJoin.get(AbstractDomainObject.UUID).in(caseUuid));
-		return em.createQuery(cq).getResultList();
-	}
+		if (caseUuids != null) {
+			List<SurveillanceReport> reports = new ArrayList<>();
 
-	public List<SurveillanceReport> getByCaseUuids(List<String> caseUuids) {
+			IterableHelper.executeBatched(caseUuids, ModelConstants.PARAMETER_LIMIT, s -> {
+				CriteriaBuilder cb = em.getCriteriaBuilder();
+				CriteriaQuery<SurveillanceReport> cq = cb.createQuery(SurveillanceReport.class);
+				Root<SurveillanceReport> reportRoot = cq.from(SurveillanceReport.class);
+				Join<SurveillanceReport, Case> caseJoin = reportRoot.join(SurveillanceReport.CAZE, JoinType.LEFT);
 
-		CriteriaBuilder cb = em.getCriteriaBuilder();
-		CriteriaQuery<SurveillanceReport> cq = cb.createQuery(SurveillanceReport.class);
-		Root<SurveillanceReport> reportRoot = cq.from(SurveillanceReport.class);
-		Join<SurveillanceReport, Case> caseJoin = reportRoot.join(SurveillanceReport.CAZE, JoinType.LEFT);
+				cq.where(caseJoin.get(AbstractDomainObject.UUID).in(caseUuids));
```

Comment:
You need to use the callback parameter here, otherwise executeBatched won't help in any way (because you're just using the initial list of UUIDs to execute the query right now)